erpnext/stock/doctype/warehouse/warehouse.py

```diff
@@ -177,7 +177,22 @@ def convert_to_group_or_ledger():
 	return frappe.get_doc("Warehouse", args.docname).convert_to_group_or_ledger()
 
 def get_child_warehouses(warehouse):
-	p_warehouse = frappe.get_doc("Warehouse", warehouse)
+	lft, rgt = frappe.get_cached_value("Warehouse", warehouse, [lft, rgt])
 
 	return frappe.db.sql_list("""select name from `tabWarehouse`
-		where lft >= %s and rgt =< %s""", (p_warehouse.lft, p_warehouse.rgt))
+		where lft >= %s and rgt =< %s""", (lft, rgt))
+
+def get_warehouses_based_on_account(account):
+	warehouses = []
+	for d in frappe.get_all("Warehouse", fields = ["name", "is_group"],
+		filters = {"account": account}):
+		if d.is_group:
+			warehouses.extend(get_child_warehouses(d.name))
+		else:
+			warehouses.append(d.name)
+
+	if not warehouses:
```

Comment:
If not warehouses and stock account is mentioned in Company, it should return all warehouses